src/server/publicApi.js

```diff
@@ -1,125 +1,181 @@
+                        }
+                    }
+                    if (cell !== null) {
+                        let threats = [];
+                        if (Array.isArray(cell.threats)) {
+                            threats = cell.threats;
+                        }
+                        Object.keys(
+                            res.G.identifiedThreats[diagramIdx][componentIdx]
+                        ).forEach((threatIdx) => {
+                            let t =
+                                res.G.identifiedThreats[diagramIdx][componentIdx][threatIdx];
+                            threats.push({
+                                status: "Open",
+                                severity: t.severity,
+                                id: t.id,
+                                methodology: "STRIDE",
+                                type: getTypeString(t.type),
+                                title: t.title,
+                                description: t.description,
+                                mitigation: t.mitigation,
+                                owner: metadata.players[t.owner].name,
+                                game: gameID,
+                            });
+                        });
+                        cell.threats = threats;
+                    }
+                }
+            );
         });
+
+        const modelTitle = model.summary.title.replace(" ", "-");
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`${modelTitle}-${timestamp}.json`);
+        ctx.body = model;
     });
 
-    const modelTitle = model.summary.title.replace(' ', '-');
-    const timestamp = (new Date()).toISOString().replace(":", "-");   
-    ctx.attachment(`${modelTitle}-${timestamp}.json`);
-    ctx.body = model;
+
+    //produce a nice textfile with the threats in
+    router.get("/download/text/:id", async (ctx) => {
+        //get some variables that might be useful
+        const gameName = ElevationOfPrivilege.name;
+        const gameID = ctx.params.id;
+        const res = await gameServer.db.get(`${gameName}:${gameID}`);
+        var threats = [];
+
+        Object.keys(res.G.identifiedThreats).forEach((diagramId) => {
+            Object.keys(res.G.identifiedThreats[diagramId]).forEach((componentId) => {
+                Object.keys(res.G.identifiedThreats[diagramId][componentId]).forEach((threatId) => {
+                    threats.push(res.G.identifiedThreats[diagramId][componentId][threatId]);
+                });
+            });
+        });
+        
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`threats-${timestamp}.txt`);
+        //ctx.body = threats;
+
+        //copied formatting from christoph's script
+        
+        ctx.body = `Threats
+=======
+
+${threats.map(
+    (threat, index) => `
+${index+1}. ${threat.title}
+${`-`.repeat(threat.title.length + 5)}
+Author: ${threat.owner}
```

Comment:
I believe the wrong owner name is provided here. If you have a look at line 114 the player id is translate via some metadata into the user name. This should also be done here.

Reply:
and then we need to test it

Reply:
I've fixed this and updated the test